docs/getting-started/certificate-pinning.adoc

```diff
@@ -0,0 +1,46 @@
+== Certificate Pinning
+
+Certificate pinning can be enabled in individual SDKs through the `mobile-services.json` file.
+Certificate pinning will only be enabled for services which are used directly by the SDKs. External
+services must have pinning configured separately. For more information on setting up certificate
+pinning for external services see the
+link:https://developer.android.com/training/articles/security-config.html#CertificatePinning[Android network security guide].
+
+=== Generating Pinning Configuration
+
+The Android SDK gets its configuration from the `https` section of the `mobile-services.json` file
+in a project.
+
+[source,json]
+----
+{
+    "services":[],
+    "https": [{
+        "host": "example.sync.service",
+        "certificateHash": "exampleHash"
+    }]
+}
+----
+
+To include the `https` section in configuration when generating it using the Mobile CLI use the
+`--include-cert-hashes` flag when retrieving a client configuration. By default, self-signed or
+invalid certs will not be permitted to be included in the certificate pinning configuration. To
+allow these to be included use the `--allow-invalid-certs` flag.
+
+[source, bash]
+----
+./mobile get clientconfig <client name> --include-cert-hashes --allow-invalid-certs
+----
+
+=== Using Pinning Configuration
+
+If the `https` section is included in the `mobile-services.json` file then certificate pinning will
+automatically be enabled in any SDK that supports it, for example the Auth SDK. Certificate pinning
+is achieved through the link:http://square.github.io/okhttp/[OkHttp HTTP client] and its
+certificate pinning implementation.
+
+=== Considerations
+
+If the certificate of a service changes then the `mobile-services.json` file will need to be
```

Comment:
"If the certificate of a service changes..." - I think this should be the Certificate Authority, rather than the Certificate since we are using SPKI pinning. The SPKI doesn't change when certs are rotated under the same CA.